serverless/aws/cdk/src/main/scala/sttp/tapir/serverless/aws/cdk/core/AppTemplateFiles.scala

```diff
@@ -0,0 +1,75 @@
+package sttp.tapir.serverless.aws.cdk.core
+
+import cats.effect._
+import cats.syntax.all._
+
+import java.io.File
+import java.nio.charset.StandardCharsets
+import java.nio.file.{Files, Paths}
+import scala.io.Source
+
+class AppTemplateFiles[F[_]: Sync](sourceDir: String, outputDir: String) {
+  private val outputStackFile = s"$outputDir/lib/tapir-cdk-stack.ts"
+  private val files = Map(
+    "bin/tapir-cdk-stack.ts" -> "bin/tapir-cdk-stack.ts",
+    "gitignore" -> ".gitignore",
+    "cdk.json" -> "cdk.json",
+    "jest.config.js" -> "jest.config.js",
+    "package.json" -> "package.json",
+    "readme.md" -> "readme.md",
+    "tsconfig.json" -> "tsconfig.json"
+  )
+
+  def copyStaticFiles(): F[Unit] =
+    files.toList.map { case (source, destination) => copy(source, destination) }.sequence.void
+
+  def clearOutputDir(): F[Unit] = deleteRecursively(Paths.get(outputDir).toFile)
+
+  def renderStackTemplate(stackTemplatePath: String, render: String => F[String]): F[Unit] =
+    for {
+      _ <- createDirectories(outputStackFile)
+      c <- getContent(stackTemplatePath)
+      r <- render(c)
+      _ <- save(r, outputStackFile)
+    } yield ()
+
+  private def getContent(path: String): F[String] =
+    Resource
+      .fromAutoCloseable[F, Source](Sync[F].blocking(Source.fromInputStream(getClass.getResourceAsStream(path))))
+      .use(content => Sync[F].delay(content.getLines().mkString(System.lineSeparator())))
+
+  private def copy(from: String, to: String): F[Unit] = {
+    val bytes = Source.fromInputStream(getClass.getResourceAsStream(sourceDir + "/" + from)).getLines().mkString("\n")
```

Comment:
this should also be surrounded by Sync[F].blocking

Reply:
fixed